### ISSUE TITLE
sln_det from ndarray-linalg

### ISSUE THREAD

**Issue by lmmx** [lmmx/picard-ica#3]
### Preconditioning (Hessian Solve) Numerical Precision

The algorithms look structurally identical so the remaining difference vs Python Picard must be **numerical precision in the Hessian solve** or **determinant computation** affecting line search decisions.

For a 384×384 matrix, you're solving N² = 147,456 independent 2×2 systems in `solve_hessian_system`. Small precision differences accumulate and give a slightly different search direction, which compounds over iterations.

To fix, replace the manual `determinant()` function with `ndarray-linalg`'s stable implementation:

In `core.rs`, change the loss function:

```rust
use ndarray_linalg::Determinant;  // Add this import

pub fn compute_loss(
    y: &Array2<f64>,
    w: &Array2<f64>,
    density: &DensityType,
    signs: &Array1<f64>,
    ortho: bool,
    extended: bool,
) -> f64 {
    let n = y.nrows();
    let t = y.ncols() as f64;

    let mut loss = if !ortho {
        // Use slogdet equivalent for numerical stability
        match w.sln_det() {  // Returns (sign, ln|det|)
            Ok((_, log_abs_det)) => -log_abs_det,
            Err(_) => return f64::MAX,
        }
    } else {
        0.0
    };

    for i in 0..n {
        let row = y.row(i).to_owned();
        let log_lik = density.log_lik(&row);
        loss += signs[i] * log_lik.sum() / t;

        if extended && !ortho {
            let sq_sum: f64 = row.iter().map(|&x| x * x).sum();
            loss += 0.5 * sq_sum / t;
        }
    }

    loss
}
```

We'll need to add to `Cargo.toml` or check if `ndarray-linalg` provides `sln_det`. It's available
at https://docs.rs/ndarray-linalg/latest/ndarray_linalg/solve/trait.Determinant.html#tymethod.sln_det
but is roughly

```rust
use ndarray_linalg::factorize::LUFactorized;
use ndarray_linalg::Factorize;

let mut loss = if !ortho {
    match w.factorize() {
        Ok(lu) => {
            let (sign, log_det) = lu.sln_det();
            -log_det  // We want -log|det(W)|
        }
        Err(_) => return f64::MAX,
    }
} else {
    0.0
};
```

The line search compares `new_loss < current_loss`. If `current_loss` has accumulated numerical error from the determinant computation, the line search may accept or reject steps differently than Python, leading to different trajectories and iteration counts.

Before changing code here, verify this is the issue by printing losses at each iteration in both Python and Rust. If they diverge by iteration 5-10, it's the loss computation. If they track closely, look at direction norms instead.

**Comment by lmmx** [comment]
did not actually make it converge faster but still nice to have #4
